File: Demoblaze_Testing_Pytest_On_BrowserStack/pages/home_page.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from config import HOME_FIRST_PRODUCT

class HomePage:
    def __init__(self, driver):
        self.driver = driver
        self.wait = WebDriverWait(driver, 20)

    def go_to_home(self):
        self.driver.get(self.driver.current_url)
        self.wait.until(EC.presence_of_element_located((By.ID, 'nava')))

    def open_first_product(self):
        el = self.wait.until(EC.element_to_be_clickable((By.XPATH, HOME_FIRST_PRODUCT)))
        el.click()

File: Demoblaze_Testing_TestNG_On_BrowserStack/src/main/java/com/example/pages/HomePage.java
package com.example.pages;

import com.example.config.Constants;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.By;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class HomePage {
    WebDriver driver;
    WebDriverWait wait;

    @FindBy(xpath = Constants.HOME_FIRST_PRODUCT)
    WebElement firstProduct;

    public HomePage(WebDriver driver) {
        this.driver = driver;
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(20));
        PageFactory.initElements(driver, this);
    }

    public void goToHome() {
        driver.get("https://www.demoblaze.com/");
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("nava")));
    }

    public void openFirstProduct() {
        wait.until(ExpectedConditions.elementToBeClickable(firstProduct));
        firstProduct.click();
    }
}
